app/jobs/flood_risk_engine/update_water_boundary_area_job.rb

```diff
@@ -0,0 +1,52 @@
+module FloodRiskEngine
+  class UpdateWaterBoundaryAreaJob < ActiveJob::Base
+    def perform(location)
+      raise MissingLocationArgumentError unless location
+      Location.transaction do
+        location.water_boundary_area = get_area_for(location)
+        location.save!
+      end
+    end
+
+    private
+
+    def get_area_for(location)
+      api_result = lookup_admin_area_using_using_external_api(location)
+
+      if inside_england?(api_result)
+        find_or_create_admin_area_from_api_result(api_result)
+      else
+        outside_england_admin_area
+      end
+    end
+
+    def lookup_admin_area_using_using_external_api(location)
+      coords = EA::AreaLookup::Coordinates.new(easting: location.easting,
+                                               northing: location.northing)
+      hash = EA::AreaLookup.find_by_coordinates(coords)
+      OpenStruct.new(hash)
+    end
+
+    def find_or_create_admin_area_from_api_result(api_result)
+      WaterBoundaryArea.find_or_create_by(code: api_result.code) do |area|
+        area.long_name = api_result.long_name
+        area.short_name = api_result.short_name
+        area.area_name = api_result.area_name
+        area.area_id = api_result.area_id
+      end
+    end
+
+    def inside_england?(api_result)
+      api_result.present? &&
+        api_result.respond_to?(:long_name) &&
+        api_result.long_name.present?
+    end
+
+    def outside_england_admin_area
+      WaterBoundaryArea.find_or_create_by(code: "outside_england") do |area|
+        area.long_name = area.short_name = area.area_name = "Outside England"
```

Comment:
given requires knowledge of 4 methods .. code maybe better within a method on WaterBoundaryArea  e.g def  build_outside_england